### PR TITLE
LogCounter

### PR DESCRIPTION
**Pull request by eliska-n**
I added the errors and warnings counter. 
I found the Logger Manager Singleton object as a perfect place for the LogCounter to live in. I just hope that the _StrucutedDataLogger covers all logging in asab. 
I'm not sure where to initialize the Counter - logging is initialized much earlier than MetricsService. I didn't want to make logging dependent on Metrics Service, so I put just a `None` placeholder to the manager and initialized the LogCounter in Native Metrics. 
Adding the logger name as a dynamic tag is quite a nice feature, isn't it? :)

### CODE REVIEW

File: asab/metrics/native.py [hunk 1]
```diff
@@ -18,6 +18,10 @@ class NativeMetrics(Service):
 
 	def __init__(self, app, metrics_svc):
 		self.MemoryGauge = metrics_svc.create_gauge("os.stat")
+
+		# injecting logging metrics into logging manager
+		logging.root.manager.LogCounter = metrics_svc.create_counter("logs", dynamic_tags=True)
```
Comment (ateska) [inline asab/metrics/native.py:23]: Let's not use dynamic tags - they have overhead (correct?)

File: asab/log.py [hunk 2]
```diff
@@ -155,6 +155,8 @@ async def schedule(app, interval):
 			level = logging.getLevelName(levelname.upper())
 			logging.getLogger(loggername).setLevel(level)
 
+		# adding placeholder for LogCounter - see native metrics
+		self.RootLogger.manager.LogCounter = None
```
Comment (ateska) [inline asab/log.py:159]: `manager`? ... is that a naming crisis?

**Review by ateska — CHANGES_REQUESTED** [review]

File: asab/log.py (hunk 2 above)
Comment (eliska-n) [reply asab/log.py:159]: That's not my manager! That's python's manager! I just used it. :D

File: asab/metrics/native.py (hunk 1 above)
Comment (eliska-n) [reply asab/metrics/native.py:23]: :( ok. it would be cool to see instantly where the logs come from.

File: asab/log.py (hunk 2 above)
Comment (ateska) [reply asab/log.py:159]: Okej ;-)

File: asab/metrics/native.py (hunk 1 above)
Comment (ateska) [reply asab/metrics/native.py:23]: This metric should be fairly simple.

**Comment by eliska-n** [comment]
It only counts logs from _StructuredDataLogger but there are more of them, especially the root one. :(

**Comment by ateska** [comment]
Then the counter is in the incorrect place.
How about artifical log target that will be added to a root logger?

**Comment by eliska-n** [comment]
Do you mean a Handler? Well... I haven't tried this, yet.

**Comment by ateska** [comment]
yes - https://docs.python.org/3/library/logging.html#logging.Logger.filter

File: asab/metrics/native.py [hunk 3]
```diff
@@ -18,6 +18,12 @@ class NativeMetrics(Service):
 
 	def __init__(self, app, metrics_svc):
 		self.MemoryGauge = metrics_svc.create_gauge("os.stat")
+
+		# Injecting logging metrics into MetricsHandler and MetricsHandler into Root Logger
+		self.MetricsLoggingHandler = MetricsLoggingHandler()
+		self.MetricsLoggingHandler.LogCounter = metrics_svc.create_counter("logs", init_values={"warnings": 0, "errors": 0}, help="Counts WARNING and ERROR logs per minute.")
```
Comment (ateska) [inline asab/metrics/native.py:24]: There are also CRITICAL levels: https://docs.python.org/3/library/logging.html#logging-levels

File: asab/metrics/native.py [hunk 4]
```diff
@@ -40,3 +46,14 @@ def _on_flushing_event(self, event_name=None):
...
 			pass
 			# L.warning("File '/proc/self/status' was not found, skipping reading native metrics.")
+
+
+class MetricsLoggingHandler(logging.Handler):
+
+	def emit(self, record):
+		level = record.levelno
+
+		if level == 30:
```
Comment (ateska) [inline asab/metrics/native.py:56]: `if level >= 30 and level < 40:` ...

**Review by ateska — CHANGES_REQUESTED** [review]

File: asab/metrics/native.py (hunk 4 above)
Comment (eliska-n) [reply asab/metrics/native.py:56]: According to the formatter it is the (25, 30> interval: https://github.com/TeskaLabs/asab/blob/master/asab/log.py#L234

Comment (ateska) [reply asab/metrics/native.py:56]: good spot, keep it aligned then, please.

**Review by ateska — APPROVED** [review]
Nice.

**Pull request merged**